### PR TITLE
Reparent tests refactoring: setup/teardown for each test, reduce cut/paste, improve readability

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**
...  and hopefully fix or isolate current flakiness

Signed-off-by: Rohit Nayak <rohit@planetscale.com>

### CODE REVIEW

File: go/test/endtoend/reparent/utils_test.go [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+
+	// Wait for mysql processes to start
+	for _, proc := range mysqlCtlProcessList {
+		if err := proc.Wait(); err != nil {
+			t.Fatalf("Error starting mysql: %s", err.Error())
+		}
+	}
+
+	// We do not need semiSync for this test case.
+	clusterInstance.EnableSemiSync = false
```
Comment (deepthi) [inline go/test/endtoend/reparent/utils_test.go:147]: VTTabletExtraArgs has `-enable_semi_sync`, whereas here it is false. Which one supersedes? Test setup should be unambiguous about it.

File: go/test/endtoend/reparent/utils_test.go [hunk 2]
```diff
@@ -0,0 +1,538 @@
+/*
```
Comment (deepthi) [inline go/test/endtoend/reparent/utils_test.go:1]: nit: utils_test -> test_utils?

**Review by deepthi — COMMENTED** [review]
This is great! Just a couple of comments.

File: go/test/endtoend/reparent/utils_test.go (hunk 1 above)
Comment (rohit-nayak-ps) [reply go/test/endtoend/reparent/utils_test.go]: Not sure what that line did, so I left it there. Removed it and the tests still run, so it seems to have been superfluous code ...

File: go/test/endtoend/reparent/utils_test.go (hunk 2 above)
Comment (rohit-nayak-ps) [reply go/test/endtoend/reparent/utils_test.go:1]: https://tip.golang.org/cmd/go/#hdr-Test_packages   
Test related files have to end with _test.go, both to access other test files in the package and so as to not get compiled into the regular binary.

**Review by deepthi — APPROVED** [review]
LGTM

**Pull request merged**